File: MvvmBase/MvvmBase_CS/ViewModel.cs
using System;

namespace CaSoftworks.MvvmBase {

    using System.ComponentModel;

    public class ViewModel: INotifyPropertyChanged {

        public event PropertyChangedEventHandler PropertyChanged;

        /// <summary>
        /// Invokes the PropertyChanged event for this class.
        /// </summary>
        /// <param name="propertyName">The name of the property that has changed.</param>
        protected void OnPropertyChanged(string propertyName) => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));

    }
}

File: MvvmBase/MvvmBase_VB/ViewModel.vb
Option Strict On

Imports System.ComponentModel

Namespace CaSoftworks.MvvmBase


    Public MustInherit Class ViewModel

        Implements INotifyPropertyChanged

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Protected Sub OnPropertyChanged(propertyName As String)

        End Sub

    End Class

End Namespace